e2e_tests/tests/experiment/test_profiling.py

```diff
@@ -0,0 +1,204 @@
+import multiprocessing as mp
+import tempfile
+from typing import Any, Dict, Optional, Sequence, Tuple
+from urllib.parse import urlencode
+
+import pytest
+import simplejson
+import yaml
+
+import determined.common.api.authentication as auth
+from determined.common import api
+from tests import config as conf
+from tests import experiment as exp
+
+
+@pytest.mark.e2e_gpu  # type: ignore
+@pytest.mark.timeout(600)  # type: ignore
+@pytest.mark.parametrize(  # type: ignore
+    "framework_base_experiment,framework_timings_enabled",
+    [
+        ("tutorials/mnist_pytorch", True),
+        ("tutorials/fashion_mnist_tf_keras", False),
+        ("computer_vision/mnist_estimator", False),
+    ],
+)
+def test_streaming_observability_metrics_apis(
+    framework_base_experiment: str, framework_timings_enabled: bool
+) -> None:
+    auth.initialize_session(conf.make_master_url(), try_reauth=True)
+
+    # To test observability, we kick off an experiment with profiling.enabled = true,
+    # then we stream the observability metric APIs.
+
+    config_path = conf.tutorials_path(f"../{framework_base_experiment}/const.yaml")
+    model_def_path = conf.tutorials_path(f"../{framework_base_experiment}")
+
+    config_obj = conf.load_config(config_path)
+    config_obj = conf.set_profiling_enabled(config_obj)
+    with tempfile.NamedTemporaryFile() as tf:
+        with open(tf.name, "w") as f:
+            yaml.dump(config_obj, f)
+        experiment_id = exp.create_experiment(
+            tf.name,
+            model_def_path,
+        )
+
+    exp.wait_for_experiment_state(experiment_id, "COMPLETED")
+    trials = exp.experiment_trials(experiment_id)
+    trial_id = trials[0]["id"]
+
+    pool = mp.pool.ThreadPool(processes=3)
```

Comment:
I would suggest using [`concurrent.futures`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.submit) to create threadpools as a context manager. 

`mp.pool.ThreadPool` predates `concurrent.futures` and the docs now [recommend using `concurrent.futures` instead](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.ThreadPool)

Reply:
Going to just remove the pool